c7n/utils.py

```diff
@@ -596,6 +604,16 @@ def format_string_values(obj, err_fallback=(IndexError, KeyError), *args, **kwar
             new.append(format_string_values(item, *args, **kwargs))
         return new
     elif isinstance(obj, str):
+        # Attempt a direct replacement rather than string
+        # interpolation if the target string consists of
+        # solely a single field replacement.
+        #
+        # This is most useful to allow non-string variables
+        # to be injected into a policy.
+        m = single_replacement.match(obj)
```

Comment:
I think this may be much simpler than dealing with a regex:

```python
    elif isinstance(obj, str):
        try:
            # Special case the "{some_key}" option where "some_key" is in the kwargs.
            # This should allow the replacement of a string with an integer.
            if len(obj) > 2 and obj[0] == "{" and obj[-1] == "}" and obj[1:-1] in kwargs:
                return kwargs[obj[1:-1]]
            return obj.format(*args, **kwargs)
        except err_fallback:
            return obj
```

Reply:
actually with the key check it does allow format strings since its pass through.

wrt to safety on callers, format_string_values gets called in quite a few places beyond policy var expansion.